example/src/main/java/org/xmtp/android/example/conversation/ConversationDetailViewModel.kt

```diff
@@ -90,4 +119,33 @@ class ConversationDetailViewModel(private val savedStateHandle: SavedStateHandle
         data class Message(override val id: String, val message: DecodedMessage) :
             MessageListItem(id, ITEM_TYPE_MESSAGE)
     }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private fun <T> Flow<T>.flowWhileShared(
+        subscriptionCount: StateFlow<Int>,
+        started: SharingStarted,
+    ): Flow<T> {
+        return started.command(subscriptionCount)
+            .distinctUntilChanged()
+            .flatMapLatest {
+                when (it) {
+                    SharingCommand.START -> this
+                    SharingCommand.STOP,
+                    SharingCommand.STOP_AND_RESET_REPLAY_CACHE,
+                    -> emptyFlow()
```

Comment:
This seems like it could use brackets since the arrow is on the next line. Just a little hard to read and understand it's for both cases